cmdx.py

```diff
@@ -1136,7 +1231,8 @@ def __repr__(self):
     def __init__(self, mobject, *args, **kwargs):
         super(DagNode, self).__init__(mobject, *args, **kwargs)
 
-        self._tfn = om.MFnTransform(mobject)
+        dagpath = om.MDagPath.getAPathTo(mobject)
+        self._tfn = om.MFnTransform(dagpath)
```

Comment:
What's happening here? Why didn't passing `mobject` suffice?

Reply:
this one is also important if you want to use MSpace.kWorld with MFnTransform functions (like translation or rotatePivot for example)
it is also needed for MFnMesh getPoints with kWorld and so on..

Reply:
Hm I see.

I would be interested in the performance impact of this, since it would affect every instantiation of nodes throughout cmdx. Could you try running a simple loop with before/after?

Have a look at [plot.py](https://github.com/mottosso/cmdx/blob/e17862ac4a5814dbed1165c839c27067338146cd/plot.py#L155) for some examples of how you can profile things like this.

If it turns out to have a significant impact, we should make it optional (somehow) since I have yet to encounter a need for these missing features and would prefer not to have to pay for it until I do.

Reply:
i agree with that. i was not comfortable with this addition. i had the feeling it could address some performance issue. i'll try to profile that from your example

a solution would be to add a function to DagNode to convert its MFnTransform

Reply:
hmm the performance loss is significant
i ran the Test of createNode transform
i have on average 416ms for cmds, 429ms for cmdx and 466ms with this modification

Reply:
I can think of two options that would suffice.

1. Lazy instantiation, for example in the `translation()`, if `kWorld` is passed, instantiate it *once* there, and then reuse it the next time around.
2. Make free functions for e.g. `cmdx.translation(node, cmdx.sWorld)` that support the features you're missing.

The latter means you pay for it only for the nodes that matter, and only when it matters, only for those that actually use it. The former means the cost is explicit. Either option seems acceptable to me.

Reply:
i like it this way and it will justify the "re-implementation" of those functions to DagNode